src/main/java/de/tum/in/www1/artemis/web/rest/AthenaResource.java

```diff
@@ -111,6 +120,21 @@ private <ExerciseT extends Exercise, SubmissionT extends Submission, OutputT> Re
         }
     }
 
+    private ResponseEntity<List<String>> getAvailableModules(long courseId, ExerciseType exerciseType) {
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        log.debug("REST request to get available Athena modules for {} exercises in course {}", exerciseType.getExerciseTypeAsReadableString(), course.getTitle());
+
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, course, null);
+
+        try {
+            List<String> modules = athenaModuleService.getAthenaModulesForCourse(course, exerciseType);
+            return ResponseEntity.ok(modules);
+        }
+        catch (NetworkingException e) {
+            throw new InternalServerErrorException("Could not fetch available Athena modules for %s exercises".formatted(exerciseType.getExerciseTypeAsReadableString()));
```

Comment:
Most of the time in the code we use string concatenation if there is only one value inserted into the string